### PR TITLE
Adding composite struct

### PR DESCRIPTION
**Pull request by asibilla**
*Add code for composite struct
*Implement composite in demo

### CODE REVIEW

File: src/structs/composite.js [hunk 1]
```diff
@@ -4,22 +4,92 @@
...
+const Member = require('./member');
+
+module.exports = class Composite {
+  constructor(data={}, config={}) {
+
+    // Cast data to the right structure.
+    data = this.validate(data) || {};
+
+    this.config = Object.assign({
+      mutable: false,
```
Comment (ktstowell) [inline src/structs/composite.js]: I'm wondering if this should default to true. Hmmm.

**Comment by ktstowell** [comment]
I'd say your demo implementation is fine. When the home component is initialized the user store should get a totally new composite instance so that any state isn't preserved.

Your update user function is essentially what the domain flow does as well so that looks good to me. I'm not convinced any abstractions would make this _cleaner_.

I'd say the only thing left is to write some tests covering read/write/remove! huzzah!

**Comment by ktstowell** [comment]
@asibilla Just making sure you see this! Github notifications are a bit weird.

**Comment by asibilla** [comment]
Hey @ktstowell, 
Still trying to back to this. I'll update the mutable property to true by default and look into writing some tests. Thanks for the input!

File: tests/unit/composite.js [hunk 2]
```diff
@@ -16,35 +17,115 @@ const Composite = using('src/structs/composite');
...
+    });
+
+    it('The key should not be changeable', () => {
+      let obj = composite.read();
+      expect(()=>{obj[key] = 'iThinkICanChangeTheKeyButICant'}).to.throw();
+    });
+
+    it('I should be able to overwrite Composite properties.', () => {
+
+      data.first_name = "Punchface";
```
Comment (ktstowell) [inline tests/unit/composite.js:71]: This is the best.

**Pull request merged**